### PR TITLE
[SPARK-19650] Commands should not trigger a Spark job

### PR DESCRIPTION
**Pull request by hvanhovell**
## What changes were proposed in this pull request?
Spark executes SQL commands eagerly. It does this by creating an RDD which contains the command's results. The downside to this is that any action on this RDD triggers a Spark job which is expensive and is unnecessary.

This PR fixes this by avoiding the materialization of an `RDD` for `Command`s; it just materializes the result and puts them in a `LocalRelation`.

## How was this patch tested?
Added a regression test to `SQLQuerySuite`.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -175,19 +175,14 @@ class Dataset[T] private[sql](
...
-    def hasSideEffects(plan: LogicalPlan): Boolean = plan match {
-      case _: Command |
-           _: InsertIntoTable => true
-      case _ => false
-    }
-
+    // For various commands (like DDL) and queries with side effects, we force query execution
+    // to happen right away to let these side effects take place eagerly.
     queryExecution.analyzed match {
       // For various commands (like DDL) and queries with side effects, we force query execution
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: remove this line

File: sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala [hunk 2]
```diff
@@ -125,8 +125,6 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
     // SHOW TABLES in Hive only output table names, while ours outputs database, table name, isTemp.
     case command: ExecutedCommandExec if command.cmd.isInstanceOf[ShowTablesCommand] =>
       command.executeCollect().map(_.getString(1))
-    case command: ExecutedCommandExec =>
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala]: why remove this case?

**Comment by cloud-fan** [comment]
LGTM

**Review by gatorsmile — APPROVED** [review]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: actually let me remove it while merging

**Comment by cloud-fan** [comment]
merging to master!

**Pull request closed without merging**